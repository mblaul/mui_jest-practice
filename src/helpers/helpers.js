import React, { Component } from 'react';

import MuiThemeProvider from '@material-ui/core/styles/MuiThemeProvider';
import theme from '../constants/theme';

import renderer from 'react-test-renderer';

export function withTheme(Component) {
  return <MuiThemeProvider theme={theme}>{Component}</MuiThemeProvider>;
}

export function createTree(wrapper) {
  return renderer.create(wrapper).toJSON();
}
