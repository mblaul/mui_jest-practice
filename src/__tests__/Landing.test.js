import React from 'react';
import { shallow, mount } from 'enzyme';
import { createShallow, createMount } from '@material-ui/core/test-utils/';

import { withTheme, createTree } from '../helpers/helpers';

import Landing from '../components/landing/Landing';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

describe('<Landing />', () => {
  let shallow;
  let shallowWrapper;
  let mount;
  let mountWrapper;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
    shallowWrapper = shallow(withTheme(<Landing />));

    mount = createMount();
    mountWrapper = mount(withTheme(<Landing />));
  });

  it('renders without crashing', () => {
    shallow(withTheme(<Landing />));
  });

  it('has a Paper element', () => {
    expect(mountWrapper.containsMatchingElement(Paper)).toEqual(true);
  });

  it('has a Typography element', () => {
    expect(createTree(mountWrapper)).toMatchSnapshot();
    expect(mountWrapper.containsMatchingElement(Typography)).toEqual(true);
  });
});
