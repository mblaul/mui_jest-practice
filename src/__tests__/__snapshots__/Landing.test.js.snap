// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> has a Typography element 1`] = `
<div>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 Landing-root-1"
  >
    <h3
      className="MuiTypography-root-29 MuiTypography-h5-45"
    >
      This is a sheet of paper.
    </h3>
    <p
      className="MuiTypography-root-29 MuiTypography-body1-38"
    >
      Paper can be used to build surface or other elements for your application.
    </p>
  </div>
</div>
`;
